File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int n = dimensions.length;
        int maxArea = 0;   // stores best area so far
        int maxDiag = 0;   // stores best diagonal^2 so far

        for (int i = 0; i < n; i++) {
            int len = dimensions[i][0];
            int w = dimensions[i][1];
            int diagSq = len * len + w * w; // diagonal squared
            int area = len * w;

            if (diagSq > maxDiag || (diagSq == maxDiag && area > maxArea)) {
                maxDiag = diagSq;
                maxArea = area;
            }
        }
        return maxArea;
    }
}

File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        double diag_length=-1;
        int area;
        for(int i=0;i<dimensions.size();i++){
                double diag=sqrt(dimensions[i][0]*dimensions[i][0]+dimensions[i][1]*dimensions[i][1]);
                if(diag>diag_length){
                    area=dimensions[i][0]*dimensions[i][1];
                    diag_length=diag;
                }
                else if(diag==diag_length){
                    area=max(area,dimensions[i][0]*dimensions[i][1]);
                }      
        }
        return area;
    }
};